core/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java

```diff
@@ -360,6 +377,25 @@ private Subchannel createMockSubchannel() {
     return subchannel;
   }
 
+  private void mockSubchannelAttributes(Subchannel subchannel, Attributes attributes) {
+    when(subchannel.getAttributes()).thenReturn(attributes);
+  }
+
+  private void mockSubchannelState(Subchannel subchannel, ConnectivityStateInfo state) {
+    mockSubchannelAttributes(subchannel, Attributes.newBuilder().set(STATE_INFO,
```

Comment:
The Attributes of a Subchannel is created more than once -- once in `createMockSubchannel()` and possibly multiple times in here. This is not right, because there should be only one Attributes instance for each Subchannel.

Actually, it should be the LB that set the STATE_INFO attribute. Setting it from outside is not expected by the LB. So for every call site of `mockSubchannelState()`, I would call `handleSubchannelState()` and leave the LB to update STATE_INFO. Granted it is a preexisting issue, but it's good to address it now.